Bump Lombok version to allow building with a JDK >=12

**Describe the bug**
Running `mvn clean install` on branch master using JDK13 results on a Lombok annotation failure.

**To Reproduce**
Steps to reproduce the behavior:
1. Pull JDK13
2. Run `mvn install`

**Expected behavior**
Build success

**Screenshots**
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:testCompile (default-testCompile) on project integration: Compilation failure
[ERROR] Lombok annotation handler class lombok.javac.handlers.HandleBuilder failed on /home/pierrez/workspace/java/pulsar/tests/integration/src/test/java/org/apache/pulsar/tests/integration/topologies/PulsarClusterSpec.java: 
java.lang.IllegalStateException: Lombok TreeMaker frontend issue: no match when looking for method: com.sun.tools.javac.tree.JCTree$JCCase com.sun.tools.javac.tree.TreeMaker.Case(com.sun.tools.javac.tree.JCTree$JCExpression, com.sun.tools.javac.util.List)
```

It is an [closed Lombok issue](rzwitserloot/lombok/issues/1888), so we should be fine by upgrading Lombok.

**Desktop (please complete the following information):**

```bash
uname -a
# Linux pierrez-xps 5.3.0-23-generic #25-Ubuntu SMP Tue Nov 12 09:22:33 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

java -version
# openjdk version "13.0.1" 2019-10-15
# OpenJDK Runtime Environment AdoptOpenJDK (build 13.0.1+9)
# OpenJDK 64-Bit Server VM AdoptOpenJDK (build 13.0.1+9, mixed mode, sharing)
```

--- a/pom.xml
+++ b/pom.xml
@@ -1050,7 +1050,7 @@ flexible messaging model and an intuitive client API.</description>
       <dependency>
         <groupId>org.projectlombok</groupId>
         <artifactId>lombok</artifactId>
-        <version>1.18.4</version>
+        <version>1.18.10</version>
       </dependency>
 
       <dependency>

